## Final Project: A Bakery in Portland, OR

### Question 1: Introduction/Business Problem

#### In this notebook, I will explore neighborhoods in Portland, Oregon in order to find a good location for a bakery/coffee shop. My target audience is people who might want to start a local business such as bakers or coffee shop owners. The location of the bakery is important to the audience because they want to place their shop in a neighborhood that does not have similar businesses, so there will be less competition. This audience also wants their bakery to be a neighborhood bakery where residents can easily pick up fresh bread and pastries, or grab coffee and walk at nearby parks. Therefore, the audeince wants the location to be residential, in a neighborhood with a few parks.

### Question 2: Data

#### I will use Foursquare data to explore the nighborhoods in Portland, OR. I will get lists of venues in each neighborhood, and look for neigborhoods where bakeries and coffee shops are not in the list of venues. From that list I will choose neighborhoods that are more residential, so they won't have a high numbr of venues, though they do want some venues, especially parks, schools, playgrounds, and maybe some restauraunts. Also the venues in th eneighborhood should not include places like airports, night clubs, shopping malls, or distribution centers, as the audience wants a quiet neighborhood bakery.

#### Before I use the Foursquare data, I will obtain data about Portland neighborhoods. My data consists of a geoJSON file from PortlandMaps.com, that shows each neighborhood and its geometry. I will use geoPandas to get the centroid of each neighborhood, and then clean the data before I start using Foursquare.

In [1]:
import pandas as pd

#### Install Folium

In [2]:
!pip install folium

#### Import Folium and make an initial map of Portland

In [2]:
import folium

m = folium.Map(location=[45.5236, -122.6750])

In [3]:
m

#### Install and import necessary libraries: requests, json, geopandas. Geopandas is a library that handles geoJSON files. I needed this because the neighborhood data from Portlland Open Maps was in GeoJSON format, and I wanted to be able put the data in a dataframe, so that I could use the Foursquare data.

In [4]:
import requests
import json
import numpy as np

In [6]:
conda install geopandas

Solving environment: done

# All requested packages already installed.


Note: you may need to restart the kernel to use updated packages.


In [5]:
import geopandas as gpd

## Obtain the Data

#### Use geoJSON file from PortlandMaps Open Data: https://gis-pdx.opendata.arcgis.com/datasets/neighborhood-boundaries
#### Shows Neighborhood boundaries. Metadata is here: https://www.portlandmaps.com/metadata/index.cfm?&action=DisplayLayer&LayerID=53509

In [6]:
URL = "https://opendata.arcgis.com/datasets/1ef75e34b8504ab9b14bef0c26cade2c_3.geojson"
gdf = gpd.read_file(URL)

gdf

OBJECTID                                 NAME COMMPLAN SHARED COALIT  \
0           1                              LINNTON               N   NWNW   
1           2                  FOREST PARK/LINNTON               Y   NWNW   
2           3                          FOREST PARK               N   NWNW   
3           4                       CATHEDRAL PARK               N   NPNS   
4           5                      UNIVERSITY PARK               N   NPNS   
..        ...                                  ...      ...    ...    ...   
125       126                               KENTON   ALBINA      N   NPNS   
126       127                            BRIDGETON               N   NPNS   
127       128                        EAST COLUMBIA               N   NPNS   
128       129  SUNDERLAND ASSOCIATION OF NEIGHBORS               N    CNN   
129       130                           PORTSMOUTH               N   NPNS   

    HORZ_VERT  Shape_Length             MAPLABEL   ID  \
0        HORZ  52741.719772              Linnton    1   
1        None  57723.635350  Forest Park/Linnton    2   
2        HORZ  82725.497522          Forest Park    3   
3        HORZ  11434.254777       Cathedral Park    4   
4        HORZ  11950.859827      University Park    5   
..        ...           ...                  ...  ...   
125      HORZ  19247.188225               Kenton  126   
126      HORZ   8635.720662            Bridgeton  127   
127      HORZ  15397.269131        East Columbia  128   
128      HORZ  20706.496916           Sunderland  129   
129      HORZ  10003.795565           Portsmouth  130   

                                              geometry  
0    POLYGON ((-122.82371 45.60616, -122.82319 45.6...  
1    POLYGON ((-122.82319 45.60616, -122.82371 45.6...  
2    POLYGON ((-122.79159 45.54843, -122.79086 45.5...  
3    POLYGON ((-122.76461 45.58519, -122.76135 45.5...  
4    POLYGON ((-122.73855 45.58395, -122.74104 45.5...  
..                                                 ...  
125  POLYGON ((-122.67859 45.57721, -122.67853 45.5...  
126  POLYGON ((-122.65704 45.60239, -122.65893 45.6...  
127  POLYGON ((-122.66015 45.59948, -122.66041 45.5...  
128  POLYGON ((-122.64031 45.60116, -122.64095 45.6...  
129  POLYGON ((-122.70795 45.58472, -122.70788 45.5...  

[130 rows x 10 columns]

#### Reset the index to make it easier to read

In [7]:
gdf = gdf.set_index("OBJECTID")
gdf

NAME COMMPLAN SHARED COALIT  \
OBJECTID                                                               
1                                     LINNTON               N   NWNW   
2                         FOREST PARK/LINNTON               Y   NWNW   
3                                 FOREST PARK               N   NWNW   
4                              CATHEDRAL PARK               N   NPNS   
5                             UNIVERSITY PARK               N   NPNS   
...                                       ...      ...    ...    ...   
126                                    KENTON   ALBINA      N   NPNS   
127                                 BRIDGETON               N   NPNS   
128                             EAST COLUMBIA               N   NPNS   
129       SUNDERLAND ASSOCIATION OF NEIGHBORS               N    CNN   
130                                PORTSMOUTH               N   NPNS   

         HORZ_VERT  Shape_Length             MAPLABEL   ID  \
OBJECTID                                                     
1             HORZ  52741.719772              Linnton    1   
2             None  57723.635350  Forest Park/Linnton    2   
3             HORZ  82725.497522          Forest Park    3   
4             HORZ  11434.254777       Cathedral Park    4   
5             HORZ  11950.859827      University Park    5   
...            ...           ...                  ...  ...   
126           HORZ  19247.188225               Kenton  126   
127           HORZ   8635.720662            Bridgeton  127   
128           HORZ  15397.269131        East Columbia  128   
129           HORZ  20706.496916           Sunderland  129   
130           HORZ  10003.795565           Portsmouth  130   

                                                   geometry  
OBJECTID                                                     
1         POLYGON ((-122.82371 45.60616, -122.82319 45.6...  
2         POLYGON ((-122.82319 45.60616, -122.82371 45.6...  
3         POLYGON ((-122.79159 45.54843, -122.79086 45.5...  
4         POLYGON ((-122.76461 45.58519, -122.76135 45.5...  
5         POLYGON ((-122.73855 45.58395, -122.74104 45.5...  
...                                                     ...  
126       POLYGON ((-122.67859 45.57721, -122.67853 45.5...  
127       POLYGON ((-122.65704 45.60239, -122.65893 45.6...  
128       POLYGON ((-122.66015 45.59948, -122.66041 45.5...  
129       POLYGON ((-122.64031 45.60116, -122.64095 45.6...  
130       POLYGON ((-122.70795 45.58472, -122.70788 45.5...  

[130 rows x 9 columns]

#### Find the centroid of each neighborhood polygon

In [8]:
gdf['centroid'] = gdf.centroid
gdf['centroid']

OBJECTID
1      POINT (-122.79326 45.60379)
2      POINT (-122.78177 45.58063)
3      POINT (-122.79208 45.56438)
4      POINT (-122.75732 45.58737)
5      POINT (-122.73008 45.57635)
                  ...             
126    POINT (-122.69739 45.59465)
127    POINT (-122.66805 45.60298)
128    POINT (-122.66187 45.59390)
129    POINT (-122.63658 45.58387)
130    POINT (-122.71838 45.58719)
Name: centroid, Length: 130, dtype: geometry

In [9]:
gdf.head(10)

NAME COMMPLAN SHARED     COALIT HORZ_VERT  \
OBJECTID                                                                        
1                                LINNTON               N       NWNW      HORZ   
2                    FOREST PARK/LINNTON               Y       NWNW      None   
3                            FOREST PARK               N       NWNW      HORZ   
4                         CATHEDRAL PARK               N       NPNS      HORZ   
5                        UNIVERSITY PARK               N       NPNS      HORZ   
6                       MC UNCLAIMED #14               N  UNCLAIMED        NA   
7                               PIEDMONT   ALBINA      N       NPNS      VERT   
8                               WOODLAWN   ALBINA      N       NECN      HORZ   
9         CULLY ASSOCIATION OF NEIGHBORS               N        CNN      HORZ   
10                           ARBOR LODGE   ALBINA      N       NPNS      HORZ   

          Shape_Length                        MAPLABEL  ID  \
OBJECTID                                                     
1         52741.719772                         Linnton   1   
2         57723.635350             Forest Park/Linnton   2   
3         82725.497522                     Forest Park   3   
4         11434.254777                  Cathedral Park   4   
5         11950.859827                 University Park   5   
6         23667.613908                MC Unclaimed #14   6   
7         10849.327392                        Piedmont   7   
8          8078.360994                        Woodlawn   8   
9         18179.392090  Cully Association of Neighbors   9   
10         9466.411504                     Arbor Lodge  10   

                                                   geometry  \
OBJECTID                                                      
1         POLYGON ((-122.82371 45.60616, -122.82319 45.6...   
2         POLYGON ((-122.82319 45.60616, -122.82371 45.6...   
3         POLYGON ((-122.79159 45.54843, -122.79086 45.5...   
4         POLYGON ((-122.76461 45.58519, -122.76135 45.5...   
5         POLYGON ((-122.73855 45.58395, -122.74104 45.5...   
6         POLYGON ((-122.76461 45.58519, -122.76542 45.5...   
7         POLYGON ((-122.67545 45.58659, -122.67593 45.5...   
8         POLYGON ((-122.66133 45.58112, -122.66136 45.5...   
9         POLYGON ((-122.62053 45.57178, -122.62052 45.5...   
10        POLYGON ((-122.67859 45.57721, -122.68210 45.5...   

                             centroid  
OBJECTID                               
1         POINT (-122.79326 45.60379)  
2         POINT (-122.78177 45.58063)  
3         POINT (-122.79208 45.56438)  
4         POINT (-122.75732 45.58737)  
5         POINT (-122.73008 45.57635)  
6         POINT (-122.72747 45.55769)  
7         POINT (-122.67042 45.57644)  
8         POINT (-122.65304 45.57257)  
9         POINT (-122.60151 45.56375)  
10        POINT (-122.69084 45.57215)

## Clean the Data

#### I have several columns that I don't need. I will drop commplan, shared, coalit, horz_vert, shape_length, maplabel, and ID

In [10]:
gdf.drop(columns=['COMMPLAN', 'SHARED', 'COALIT', 'HORZ_VERT', 'Shape_Length', 'MAPLABEL', 'ID'], inplace = True)
gdf.head()

NAME  \
OBJECTID                        
1                     LINNTON   
2         FOREST PARK/LINNTON   
3                 FOREST PARK   
4              CATHEDRAL PARK   
5             UNIVERSITY PARK   

                                                   geometry  \
OBJECTID                                                      
1         POLYGON ((-122.82371 45.60616, -122.82319 45.6...   
2         POLYGON ((-122.82319 45.60616, -122.82371 45.6...   
3         POLYGON ((-122.79159 45.54843, -122.79086 45.5...   
4         POLYGON ((-122.76461 45.58519, -122.76135 45.5...   
5         POLYGON ((-122.73855 45.58395, -122.74104 45.5...   

                             centroid  
OBJECTID                               
1         POINT (-122.79326 45.60379)  
2         POINT (-122.78177 45.58063)  
3         POINT (-122.79208 45.56438)  
4         POINT (-122.75732 45.58737)  
5         POINT (-122.73008 45.57635)

#### I need to convert te centroid to a lat/long column, but its type is geometry. I'll convert it to a string so it's a little eaier to work with

In [11]:
gdf.dtypes

NAME          object
geometry    geometry
centroid    geometry
dtype: object

In [12]:
gdf["centroid"]=gdf["centroid"].astype("str")

In [13]:
gdf.dtypes

NAME          object
geometry    geometry
centroid      object
dtype: object

#### Remove the word "point" from the centroid column

In [14]:
gdf['centroid'] = gdf['centroid'].str.replace('POINT ', '')

In [15]:
gdf.head()

NAME  \
OBJECTID                        
1                     LINNTON   
2         FOREST PARK/LINNTON   
3                 FOREST PARK   
4              CATHEDRAL PARK   
5             UNIVERSITY PARK   

                                                   geometry  \
OBJECTID                                                      
1         POLYGON ((-122.82371 45.60616, -122.82319 45.6...   
2         POLYGON ((-122.82319 45.60616, -122.82371 45.6...   
3         POLYGON ((-122.79159 45.54843, -122.79086 45.5...   
4         POLYGON ((-122.76461 45.58519, -122.76135 45.5...   
5         POLYGON ((-122.73855 45.58395, -122.74104 45.5...   

                                        centroid  
OBJECTID                                          
1         (-122.7932636868726 45.60378993637875)  
2         (-122.7817746512875 45.58063023423158)  
3         (-122.7920776241818 45.56438278026928)  
4         (-122.7573167006587 45.58736826406709)  
5          (-122.730079200974 45.57635375668902)

#### Now I need to separate the centroid into two separate columns:  latitude and longitude. This will allow me to map the center of each Portland neighborhood.

In [16]:
# Create two empty lists for the results
Latitude = []
Longitude = []

# For each row in centroid
for row in gdf['centroid']:
    # Split the row by the space and append
    # everything before the space to longitude
    Longitude.append(row.split(' ')[0])
    # Split the row by the space and append
    # everything after the space to latitude
    Latitude.append(row.split(' ')[1])
    
gdf['Latitude'] = Latitude
gdf['Longitude'] = Longitude

In [17]:
gdf.head()

NAME  \
OBJECTID                        
1                     LINNTON   
2         FOREST PARK/LINNTON   
3                 FOREST PARK   
4              CATHEDRAL PARK   
5             UNIVERSITY PARK   

                                                   geometry  \
OBJECTID                                                      
1         POLYGON ((-122.82371 45.60616, -122.82319 45.6...   
2         POLYGON ((-122.82319 45.60616, -122.82371 45.6...   
3         POLYGON ((-122.79159 45.54843, -122.79086 45.5...   
4         POLYGON ((-122.76461 45.58519, -122.76135 45.5...   
5         POLYGON ((-122.73855 45.58395, -122.74104 45.5...   

                                        centroid            Latitude  \
OBJECTID                                                               
1         (-122.7932636868726 45.60378993637875)  45.60378993637875)   
2         (-122.7817746512875 45.58063023423158)  45.58063023423158)   
3         (-122.7920776241818 45.56438278026928)  45.56438278026928)   
4         (-122.7573167006587 45.58736826406709)  45.58736826406709)   
5          (-122.730079200974 45.57635375668902)  45.57635375668902)   

                    Longitude  
OBJECTID                       
1         (-122.7932636868726  
2         (-122.7817746512875  
3         (-122.7920776241818  
4         (-122.7573167006587  
5          (-122.730079200974

#### Take those parentheses off the lat and long columns

In [18]:
gdf['Latitude'] = gdf['Latitude'].str.replace(')', '')

In [19]:
gdf['Longitude'] = gdf['Longitude'].str.replace('(', '')

In [20]:
gdf.head()

NAME  \
OBJECTID                        
1                     LINNTON   
2         FOREST PARK/LINNTON   
3                 FOREST PARK   
4              CATHEDRAL PARK   
5             UNIVERSITY PARK   

                                                   geometry  \
OBJECTID                                                      
1         POLYGON ((-122.82371 45.60616, -122.82319 45.6...   
2         POLYGON ((-122.82319 45.60616, -122.82371 45.6...   
3         POLYGON ((-122.79159 45.54843, -122.79086 45.5...   
4         POLYGON ((-122.76461 45.58519, -122.76135 45.5...   
5         POLYGON ((-122.73855 45.58395, -122.74104 45.5...   

                                        centroid           Latitude  \
OBJECTID                                                              
1         (-122.7932636868726 45.60378993637875)  45.60378993637875   
2         (-122.7817746512875 45.58063023423158)  45.58063023423158   
3         (-122.7920776241818 45.56438278026928)  45.56438278026928   
4         (-122.7573167006587 45.58736826406709)  45.58736826406709   
5          (-122.730079200974 45.57635375668902)  45.57635375668902   

                   Longitude  
OBJECTID                      
1         -122.7932636868726  
2         -122.7817746512875  
3         -122.7920776241818  
4         -122.7573167006587  
5          -122.730079200974

#### I'm converting the gdf to a df, dropping the geometry and centroid colums, but saving the variable in case I need it later to map the neighborhoods

In [21]:
from pandas import DataFrame

pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

from geopy.geocoders import Nominatim # convert an address into latitude and longitude values

import requests # library to handle requests
from pandas.io.json import json_normalize # tranform JSON file into a pandas dataframe

# Matplotlib and associated plotting modules
import matplotlib.cm as cm
import matplotlib.colors as colors

# import k-means from clustering stage
from sklearn.cluster import KMeans

print('Libraries imported.')

Libraries imported.


In [22]:
gdf.drop(columns=['geometry', 'centroid'], inplace = True)
gdf.head()

NAME           Latitude           Longitude
OBJECTID                                                            
1                     LINNTON  45.60378993637875  -122.7932636868726
2         FOREST PARK/LINNTON  45.58063023423158  -122.7817746512875
3                 FOREST PARK  45.56438278026928  -122.7920776241818
4              CATHEDRAL PARK  45.58736826406709  -122.7573167006587
5             UNIVERSITY PARK  45.57635375668902   -122.730079200974

In [23]:
df = pd.DataFrame(gdf)

#### Convert lat and long to floats

In [24]:
df.dtypes

NAME         object
Latitude     object
Longitude    object
dtype: object

In [25]:
df["Latitude"]=df["Latitude"].astype(float)

In [26]:
df["Longitude"]=df["Longitude"].astype(float)

In [27]:
df.dtypes

NAME          object
Latitude     float64
Longitude    float64
dtype: object

In [28]:
df.head(10)

NAME   Latitude   Longitude
OBJECTID                                                       
1                                LINNTON  45.603790 -122.793264
2                    FOREST PARK/LINNTON  45.580630 -122.781775
3                            FOREST PARK  45.564383 -122.792078
4                         CATHEDRAL PARK  45.587368 -122.757317
5                        UNIVERSITY PARK  45.576354 -122.730079
6                       MC UNCLAIMED #14  45.557689 -122.727474
7                               PIEDMONT  45.576438 -122.670418
8                               WOODLAWN  45.572565 -122.653037
9         CULLY ASSOCIATION OF NEIGHBORS  45.563753 -122.601509
10                           ARBOR LODGE  45.572152 -122.690842

In [29]:
df.reset_index(drop=True, inplace=True)
df.head()


NAME   Latitude   Longitude
0              LINNTON  45.603790 -122.793264
1  FOREST PARK/LINNTON  45.580630 -122.781775
2          FOREST PARK  45.564383 -122.792078
3       CATHEDRAL PARK  45.587368 -122.757317
4      UNIVERSITY PARK  45.576354 -122.730079

### Show a map of Portland with centroids of each neighborhood

In [30]:
# create map of Portland Neighborhoods using latitude and longitude values
m = folium.Map(location=[45.5236, -122.6750], zoom_start=12)

# add markers to map
for lat, lng, label in zip(df['Latitude'], df['Longitude'], df['NAME']):
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(m)  
    
m

## Use Foursquare to explore neighborhoods

#### Enter my info for foursquare

In [31]:
CLIENT_ID = 'TDKMGX0URSB410BSSHFVA2UOSPJGWPVY5MUA3ME5APTIGSTS' # your Foursquare ID
CLIENT_SECRET = 'LDINZ0KVRKCBSRBLXWE0WM2CGFFHHX3MDFLZBEGAGA5WFKKF' # your Foursquare Secret
VERSION = '20180605' # Foursquare API version
LIMIT = 100 # A default Foursquare API limit value

#### Here we definte a function that will get venues within 500 m of my neighborhood centroids

In [32]:
def getNearbyVenues(names, latitudes, longitudes, radius=500):
    
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
            
        # make the GET request
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        # return only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Neighborhood', 
                  'Neighborhood Latitude', 
                  'Neighborhood Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    
    return(nearby_venues)

#### Run the "getNearbyVenues" function on each Portland neighborhood. Then print the size and show the first 20 rows

In [33]:
Portland_venues = getNearbyVenues(names=df['NAME'],
                                   latitudes=df['Latitude'],
                                   longitudes=df['Longitude']
                                  )

LINNTON
FOREST PARK/LINNTON
FOREST PARK
CATHEDRAL PARK
UNIVERSITY PARK
MC UNCLAIMED #14
PIEDMONT
WOODLAWN
CULLY ASSOCIATION OF NEIGHBORS
ARBOR LODGE
OVERLOOK
CONCORDIA
PARKROSE
SUMNER ASSOCIATION OF NEIGHBORS
ARGAY TERRACE
HUMBOLDT
KING
VERNON
WILKES COMMUNITY GROUP
BEAUMONT-WILSHIRE
SABIN COMMUNITY ASSOCIATION
ALAMEDA
BOISE
NORTHWEST HEIGHTS
ROSEWAY
MADISON SOUTH
ARGAY/WILKES COMMUNITY GROUP
BOISE/ELIOT
ELIOT
IRVINGTON COMMUNITY ASSOCIATION
SABIN COMMUNITY ASSN./IRVINGTON COMMUNITY ASSN.
ALAMEDA/IRVINGTON COMMUNITY ASSN.
ROSE CITY PARK
PARKROSE HEIGHTS ASSOCIATION OF NEIGHBORS
NORTHWEST DISTRICT ASSOCIATION
ALAMEDA/BEAUMONT-WILSHIRE
FOREST PARK/NORTHWEST DISTRICT ASSOCIATION
RUSSELL
ROSEWAY/MADISON SOUTH
GRANT PARK
MC UNCLAIMED #5
PEARL DISTRICT
GRANT PARK/HOLLYWOOD
HOLLYWOOD
WOODLAND PARK
LLOYD DISTRICT COMMUNITY ASSOCIATION
SULLIVAN'S GULCH
SULLIVAN'S GULCH/GRANT PARK
MONTAVILLA
HILLSIDE/NORTHWEST DISTRICT ASSN.
LAURELHURST
KERNS
LLOYD DISTRICT COMMUNITY ASSN./SULLIVAN'S GULCH
HILLS

In [64]:
print(Portland_venues.shape)
Portland_venues.head(20)

(2003, 6)


Neighborhood Latitude  Neighborhood Longitude  \
Neighborhood                                                         
LINNTON                          45.603790             -122.793264   
LINNTON                          45.603790             -122.793264   
LINNTON                          45.603790             -122.793264   
FOREST PARK/LINNTON              45.580630             -122.781775   
CATHEDRAL PARK                   45.587368             -122.757317   
CATHEDRAL PARK                   45.587368             -122.757317   
CATHEDRAL PARK                   45.587368             -122.757317   
CATHEDRAL PARK                   45.587368             -122.757317   
CATHEDRAL PARK                   45.587368             -122.757317   
CATHEDRAL PARK                   45.587368             -122.757317   
CATHEDRAL PARK                   45.587368             -122.757317   
CATHEDRAL PARK                   45.587368             -122.757317   
CATHEDRAL PARK                   45.587368             -122.757317   
CATHEDRAL PARK                   45.587368             -122.757317   
CATHEDRAL PARK                   45.587368             -122.757317   
CATHEDRAL PARK                   45.587368             -122.757317   
CATHEDRAL PARK                   45.587368             -122.757317   
CATHEDRAL PARK                   45.587368             -122.757317   
CATHEDRAL PARK                   45.587368             -122.757317   
CATHEDRAL PARK                   45.587368             -122.757317   

                                               Venue  Venue Latitude  \
Neighborhood                                                           
LINNTON                                       Subway       45.603014   
LINNTON                                        Shell       45.603473   
LINNTON                                     7-Eleven       45.602668   
FOREST PARK/LINNTON   Forest Park Hardesty Trailhead       45.578979   
CATHEDRAL PARK                        Cathedral Park       45.587744   
CATHEDRAL PARK                        Hoplandia Beer       45.589662   
CATHEDRAL PARK                  Occidental Wursthaus       45.588864   
CATHEDRAL PARK       Taqueria Y Panaderia Santa Cruz       45.590201   
CATHEDRAL PARK                       The Great North       45.590399   
CATHEDRAL PARK            Occidental Brewing Company       45.588807   
CATHEDRAL PARK                      Homegrown Smoker       45.590409   
CATHEDRAL PARK          Blue Moon Camera and Machine       45.589724   
CATHEDRAL PARK               St Johns Farmers Market       45.589882   
CATHEDRAL PARK              St Johns Coffee Roasters       45.589342   
CATHEDRAL PARK                StormBreaker St. Johns       45.589758   
CATHEDRAL PARK                          Thai Cottage       45.590203   
CATHEDRAL PARK                              Affogato       45.590690   
CATHEDRAL PARK                           Burgerville       45.589892   
CATHEDRAL PARK                     Two Stroke Coffee       45.591680   
CATHEDRAL PARK         St. Johns Twin Cinema and Pub       45.590679   

                     Venue Longitude                 Venue Category  
Neighborhood                                                         
LINNTON                  -122.788590                 Sandwich Place  
LINNTON                  -122.788861                    Gas Station  
LINNTON                  -122.788165              Convenience Store  
FOREST PARK/LINNTON      -122.781064                          Trail  
CATHEDRAL PARK           -122.759822                           Park  
CATHEDRAL PARK           -122.755614                     Beer Store  
CATHEDRAL PARK           -122.761344              German Restaurant  
CATHEDRAL PARK           -122.755332             Mexican Restaurant  
CATHEDRAL PARK           -122.754684                    Coffee Shop  
CATHEDRAL PARK           -122.761680                        Brewery  
CATHEDRAL PARK           -122.755313  Vegetarian / Vega

In [65]:
Portland_venues.reset_index(level =['Neighborhood'], inplace = True) 
Portland_venues.head()

Neighborhood  Neighborhood Latitude  Neighborhood Longitude  \
0              LINNTON              45.603790             -122.793264   
1              LINNTON              45.603790             -122.793264   
2              LINNTON              45.603790             -122.793264   
3  FOREST PARK/LINNTON              45.580630             -122.781775   
4       CATHEDRAL PARK              45.587368             -122.757317   

                            Venue  Venue Latitude  Venue Longitude  \
0                          Subway       45.603014      -122.788590   
1                           Shell       45.603473      -122.788861   
2                        7-Eleven       45.602668      -122.788165   
3  Forest Park Hardesty Trailhead       45.578979      -122.781064   
4                  Cathedral Park       45.587744      -122.759822   

      Venue Category  
0     Sandwich Place  
1        Gas Station  
2  Convenience Store  
3              Trail  
4               Park

#### How many venues for each neighborhood

In [66]:
Portland_venues.groupby('Neighborhood').count()

Neighborhood Latitude  \
Neighborhood                                                                
ALAMEDA                                                                 4   
ALAMEDA/BEAUMONT-WILSHIRE                                               4   
ALAMEDA/IRVINGTON COMMUNITY ASSN.                                       6   
ARBOR LODGE                                                            11   
ARDENWALD-JOHNSON CREEK                                                 9   
ARDENWALD-JOHNSON CREEK/WOODSTOCK                                       6   
ARGAY TERRACE                                                          12   
ARGAY/WILKES COMMUNITY GROUP                                            5   
ARLINGTON HEIGHTS                                                      21   
ARLINGTON HEIGHTS/SYLVAN-HIGHLANDS                                     22   
ARNOLD CREEK                                                            2   
ASHCREEK                                                                6   
ASHCREEK/CRESTWOOD                                                      6   
BEAUMONT-WILSHIRE                                                      15   
BOISE                                                                  83   
BOISE/ELIOT                                                            21   
BRENTWOOD-DARLINGTON                                                    4   
BRIDGETON                                                               4   
BRIDLEMILE                                                              3   
BRIDLEMILE/SOUTHWEST HILLS RESIDENTIAL LEAGUE                           1   
BROOKLYN ACTION CORPS                                                   8   
BUCKMAN COMMUNITY ASSOCIATION                                          34   
CATHEDRAL PARK                                                         35   
CENTENNIAL COMMUNITY ASSN./PLEASANT VALLEY                              7   
CENTENNIAL COMMUNITY ASSOCIATION                                       10   
COLLINS VIEW                                                            2   
CONCORDIA                                                              37   
CRESTON-KENILWORTH                                                     26   
CRESTWOOD                                                               2   
CULLY ASSOCIATION OF NEIGHBORS                                          5   
EAST COLUMBIA                                                           2   
EASTMORELAND                                                            1   
EASTMORELAND/ARDENWALD-JOHNSON CREEK                                    6   
EASTMORELAND/REED                                                       5   
ELIOT                                                                  19   
FAR SOUTHWEST                                                           4   
FOREST PARK/LINNTON                                                     1   
FOREST PARK/NORTHWEST DISTRICT ASSOCIATION                              4   
FOSTER-POWELL                                                           5   
GLENFAIR                                                                1   
GOOSE HOLLOW FOOTHILLS LEAGUE                                          32   
GOOSE HOLLOW FOOTHILLS LEAGUE/SOUTHWEST HILLS R...                     16   
GRANT PARK                                                              4   
GRANT PARK/HOLLYWOOD                                                   46   
HAYHURST                                                                3   
HAZELWOOD                                                              20   
HAZELWOOD/MILL PARK                                                    25   
HEALY HEIGHTS/SOUTHWEST HILLS RESIDENTIAL LEAGUE                        2   
HILLSDALE                                                              32   
HILLSIDE                                                                9   
HILLSIDE/NORTHWEST DISTRICT ASSN.               

#### I want to use only neighborhoods that are more residential, so they will have fewer venues. So I am finding all neighborhoods that have more than 15 venues, and then I will remove those from the Portland_venues df.

In [67]:
venue_totals = Portland_venues.groupby('Neighborhood').count()

In [68]:
venue_totals_filtered = venue_totals[venue_totals['Venue'] >= 15] 
venue_totals_filtered.head()

Neighborhood Latitude  \
Neighborhood                                                
ARLINGTON HEIGHTS                                      21   
ARLINGTON HEIGHTS/SYLVAN-HIGHLANDS                     22   
BEAUMONT-WILSHIRE                                      15   
BOISE                                                  83   
BOISE/ELIOT                                            21   

                                    Neighborhood Longitude  Venue  \
Neighborhood                                                        
ARLINGTON HEIGHTS                                       21     21   
ARLINGTON HEIGHTS/SYLVAN-HIGHLANDS                      22     22   
BEAUMONT-WILSHIRE                                       15     15   
BOISE                                                   83     83   
BOISE/ELIOT                                             21     21   

                                    Venue Latitude  Venue Longitude  \
Neighborhood                                                          
ARLINGTON HEIGHTS                               21               21   
ARLINGTON HEIGHTS/SYLVAN-HIGHLANDS              22               22   
BEAUMONT-WILSHIRE                               15               15   
BOISE                                           83               83   
BOISE/ELIOT                                     21               21   

                                    Venue Category  
Neighborhood                                        
ARLINGTON HEIGHTS                               21  
ARLINGTON HEIGHTS/SYLVAN-HIGHLANDS              22  
BEAUMONT-WILSHIRE                               15  
BOISE                                           83  
BOISE/ELIOT                                     21

#### Looks like I have 41 neighborhoods with 15 or more venues to remove. I'll get a list of them so I can copy-paste and then drop them.

In [69]:
venue_totals_filtered.shape

(41, 6)

#### Put the busy neighborhoods into an array

In [70]:
busy_neighborhoods = venue_totals_filtered.index.values
busy_neighborhoods

array(['ARLINGTON HEIGHTS', 'ARLINGTON HEIGHTS/SYLVAN-HIGHLANDS',
       'BEAUMONT-WILSHIRE', 'BOISE', 'BOISE/ELIOT',
       'BUCKMAN COMMUNITY ASSOCIATION', 'CATHEDRAL PARK', 'CONCORDIA',
       'CRESTON-KENILWORTH', 'ELIOT', 'GOOSE HOLLOW FOOTHILLS LEAGUE',
       'GOOSE HOLLOW FOOTHILLS LEAGUE/SOUTHWEST HILLS RESIDENTIAL LEAGUE',
       'GRANT PARK/HOLLYWOOD', 'HAZELWOOD', 'HAZELWOOD/MILL PARK',
       'HILLSDALE', 'HOLLYWOOD',
       'HOSFORD-ABERNETHY NEIGHBORHOOD DISTRICT ASSN.', 'HUMBOLDT',
       'KERNS', 'KING', "LLOYD DISTRICT COMMUNITY ASSN./SULLIVAN'S GULCH",
       'LLOYD DISTRICT COMMUNITY ASSOCIATION', 'MONTAVILLA', 'MULTNOMAH',
       'NORTHWEST DISTRICT ASSOCIATION', 'OLD TOWN COMMUNITY ASSOCIATION',
       'OVERLOOK', 'PEARL DISTRICT', 'PORTLAND DOWNTOWN', 'RICHMOND',
       'ROSE CITY PARK', 'ROSEWAY',
       'SABIN COMMUNITY ASSN./IRVINGTON COMMUNITY ASSN.',
       'SELLWOOD-MORELAND IMPROVEMENT LEAGUE', "SULLIVAN'S GULCH",
       "SULLIVAN'S GULCH/GRANT PARK", 'SUN

In [72]:
# Thinking I don't need this...
Portland_venues = Portland_venues.set_index("Neighborhood")

In [73]:
portland_venues_low = Portland_venues.drop(['ARLINGTON HEIGHTS', 'ARLINGTON HEIGHTS/SYLVAN-HIGHLANDS',
       'BEAUMONT-WILSHIRE', 'BOISE', 'BOISE/ELIOT',
       'BUCKMAN COMMUNITY ASSOCIATION', 'CATHEDRAL PARK', 'CONCORDIA',
       'CRESTON-KENILWORTH', 'ELIOT', 'GOOSE HOLLOW FOOTHILLS LEAGUE',
       'GOOSE HOLLOW FOOTHILLS LEAGUE/SOUTHWEST HILLS RESIDENTIAL LEAGUE',
       'GRANT PARK/HOLLYWOOD', 'HAZELWOOD', 'HAZELWOOD/MILL PARK',
       'HILLSDALE', 'HOLLYWOOD',
       'HOSFORD-ABERNETHY NEIGHBORHOOD DISTRICT ASSN.', 'HUMBOLDT',
       'KERNS', 'KING', "LLOYD DISTRICT COMMUNITY ASSN./SULLIVAN'S GULCH",
       'LLOYD DISTRICT COMMUNITY ASSOCIATION', 'MONTAVILLA', 'MULTNOMAH',
       'NORTHWEST DISTRICT ASSOCIATION', 'OLD TOWN COMMUNITY ASSOCIATION',
       'OVERLOOK', 'PEARL DISTRICT', 'PORTLAND DOWNTOWN', 'RICHMOND',
       'ROSE CITY PARK', 'ROSEWAY',
       'SABIN COMMUNITY ASSN./IRVINGTON COMMUNITY ASSN.',
       'SELLWOOD-MORELAND IMPROVEMENT LEAGUE', "SULLIVAN'S GULCH",
       "SULLIVAN'S GULCH/GRANT PARK", 'SUNNYSIDE', 'VERNON',
       'WOODLAND PARK', 'WOODSTOCK'], axis=0)

In [74]:
portland_venues_low.shape

(426, 6)

In [75]:
portland_venues_low

Neighborhood Latitude  \
Neighborhood                                                                
LINNTON                                                         45.603790   
LINNTON                                                         45.603790   
LINNTON                                                         45.603790   
FOREST PARK/LINNTON                                             45.580630   
UNIVERSITY PARK                                                 45.576354   
UNIVERSITY PARK                                                 45.576354   
UNIVERSITY PARK                                                 45.576354   
UNIVERSITY PARK                                                 45.576354   
MC UNCLAIMED #14                                                45.557689   
PIEDMONT                                                        45.576438   
PIEDMONT                                                        45.576438   
PIEDMONT                                                        45.576438   
PIEDMONT                                                        45.576438   
PIEDMONT                                                        45.576438   
PIEDMONT                                                        45.576438   
PIEDMONT                                                        45.576438   
PIEDMONT                                                        45.576438   
PIEDMONT                                                        45.576438   
PIEDMONT                                                        45.576438   
PIEDMONT                                                        45.576438   
PIEDMONT                                                        45.576438   
WOODLAWN                                                        45.572565   
WOODLAWN                                                        45.572565   
WOODLAWN                                                        45.572565   
WOODLAWN                                                        45.572565   
WOODLAWN                                                        45.572565   
WOODLAWN                                                        45.572565   
WOODLAWN                                                        45.572565   
WOODLAWN                                                        45.572565   
WOODLAWN                                                        45.572565   
WOODLAWN                                                        45.572565   
WOODLAWN                                                        45.572565   
WOODLAWN                                                        45.572565   
WOODLAWN                                                        45.572565   
WOODLAWN                                                        45.572565   
CULLY ASSOCIATION OF NEIGHBORS                                  45.563753   
CULLY ASSOCIATION OF NEIGHBORS                                  45.563753   
CULLY ASSOCIATION OF NEIGHBORS                                  45.563753   
CULLY ASSOCIATION OF NEIGHBORS                                  45.563753   
CULLY ASSOCIATION OF NEIGHBORS                                  45.563753   
ARBOR LODGE                                                     45.572152   
ARBOR LODGE                                                     45.572152   
ARBOR LODGE                                                     45.572152   
ARBOR LODGE                                                     45.572152   
ARBOR LODGE                                                     45.572152   
ARBOR LODGE                                                     45.572152   
ARBOR LODGE                                                     45.572152   
ARBOR LODGE                                                     45.572152   
ARBOR LODGE                                                     45.572152   
ARBOR LODGE                                                     45.572152   
ARBOR LODGE                                     

#### I've cut out the busiest neighborhoods, and now have 79 neighborhoods remianing. I'd like to remove neighborhoods with coffee shops or bakeries, as I want to find a place with less competition

In [76]:
#remove coffee shops
coffee_shops = portland_venues_low.loc[portland_venues_low['Venue Category'] == 'Coffee Shop']
coffee_shops

Neighborhood Latitude  \
Neighborhood                                               
PIEDMONT                                       45.576438   
WOODLAWN                                       45.572565   
ARBOR LODGE                                    45.572152   
ALAMEDA                                        45.549062   
ALAMEDA/IRVINGTON COMMUNITY ASSN.              45.545149   
RUSSELL                                        45.539319   
ROSEWAY/MADISON SOUTH                          45.541872   
NORTH TABOR                                    45.526056   
NORTH TABOR                                    45.526056   
SOUTH PORTLAND                                 45.488039   
HOMESTEAD                                      45.495087   
POWELLHURST-GILBERT                            45.493472   
POWELLHURST-GILBERT                            45.493472   
MT. SCOTT-ARLETA                               45.481694   
MAPLEWOOD                                      45.471896   
ARDENWALD-JOHNSON CREEK                        45.456025   

                                   Neighborhood Longitude  \
Neighborhood                                                
PIEDMONT                                      -122.670418   
WOODLAWN                                      -122.653037   
ARBOR LODGE                                   -122.690842   
ALAMEDA                                       -122.635862   
ALAMEDA/IRVINGTON COMMUNITY ASSN.             -122.641661   
RUSSELL                                       -122.527556   
ROSEWAY/MADISON SOUTH                         -122.581062   
NORTH TABOR                                   -122.605330   
NORTH TABOR                                   -122.605330   
SOUTH PORTLAND                                -122.673813   
HOMESTEAD                                     -122.687235   
POWELLHURST-GILBERT                           -122.540011   
POWELLHURST-GILBERT                           -122.540011   
MT. SCOTT-ARLETA                              -122.591914   
MAPLEWOOD                                     -122.733599   
ARDENWALD-JOHNSON CREEK                       -122.628427   

                                                           Venue  \
Neighborhood                                                       
PIEDMONT                                   Black Rock Coffee Bar   
WOODLAWN                              Woodlawn Coffee and Pastry   
ARBOR LODGE                                    Grindhouse Coffee   
ALAMEDA                                                  Guilder   
ALAMEDA/IRVINGTON COMMUNITY ASSN.                        Guilder   
RUSSELL                                               Spinal Tap   
ROSEWAY/MADISON SOUTH                              Bebo's Coffee   
NORTH TABOR                        Seven Virtues Coffee Roasters   
NORTH TABOR                                            Starbucks   
SOUTH PORTLAND                              Essence Coffee & Tea   
HOMESTEAD                                              Starbucks   
POWELLHURST-GILBERT                                    Starbucks   
POWELLHURST-GILBERT                                        Awake   
MT. SCOTT-ARLETA                             Space Monkey Coffee   
MAPLEWOOD                                 Maplewood Coffee & Tea   
ARDENWALD-JOHNSON CREEK                          Milwaukie Grind   

                                   Venue Latitude  Venue Longitude  \
Neighborhood                                                         
PIEDMONT                                45.577042      -122.668155   
WOODLAWN                                45.571753      -122.657063   
ARBOR LODGE                             45.569870      -122.687110   
ALAMEDA                                 45.548290      -122.641335   
ALAMEDA/IRVINGTON COMMUNITY ASSN.       45.548290      -122.641335   
RUSSELL                                 45.543268      -122.526020   
ROSEWAY/MADISON SOUTH                   45.545176      -122.579045   


In [77]:
#get list of neighborhoods with coffee shops
remove_coffee_shops = coffee_shops.index.values
remove_coffee_shops

array(['PIEDMONT', 'WOODLAWN', 'ARBOR LODGE', 'ALAMEDA',
       'ALAMEDA/IRVINGTON COMMUNITY ASSN.', 'RUSSELL',
       'ROSEWAY/MADISON SOUTH', 'NORTH TABOR', 'NORTH TABOR',
       'SOUTH PORTLAND', 'HOMESTEAD', 'POWELLHURST-GILBERT',
       'POWELLHURST-GILBERT', 'MT. SCOTT-ARLETA', 'MAPLEWOOD',
       'ARDENWALD-JOHNSON CREEK'], dtype=object)

In [78]:
# drop rows with coffee shops
portland_venues_low = portland_venues_low.drop(['PIEDMONT', 'WOODLAWN', 'ARBOR LODGE', 'ALAMEDA',
       'ALAMEDA/IRVINGTON COMMUNITY ASSN.', 'RUSSELL',
       'ROSEWAY/MADISON SOUTH', 'NORTH TABOR', 'NORTH TABOR',
       'SOUTH PORTLAND', 'HOMESTEAD', 'POWELLHURST-GILBERT',
       'POWELLHURST-GILBERT', 'MT. SCOTT-ARLETA', 'MAPLEWOOD',
       'ARDENWALD-JOHNSON CREEK'], axis=0)

In [79]:
portland_venues_low

Neighborhood Latitude  \
Neighborhood                                                                
LINNTON                                                         45.603790   
LINNTON                                                         45.603790   
LINNTON                                                         45.603790   
FOREST PARK/LINNTON                                             45.580630   
UNIVERSITY PARK                                                 45.576354   
UNIVERSITY PARK                                                 45.576354   
UNIVERSITY PARK                                                 45.576354   
UNIVERSITY PARK                                                 45.576354   
MC UNCLAIMED #14                                                45.557689   
CULLY ASSOCIATION OF NEIGHBORS                                  45.563753   
CULLY ASSOCIATION OF NEIGHBORS                                  45.563753   
CULLY ASSOCIATION OF NEIGHBORS                                  45.563753   
CULLY ASSOCIATION OF NEIGHBORS                                  45.563753   
CULLY ASSOCIATION OF NEIGHBORS                                  45.563753   
PARKROSE                                                        45.560177   
PARKROSE                                                        45.560177   
PARKROSE                                                        45.560177   
PARKROSE                                                        45.560177   
PARKROSE                                                        45.560177   
PARKROSE                                                        45.560177   
PARKROSE                                                        45.560177   
PARKROSE                                                        45.560177   
PARKROSE                                                        45.560177   
PARKROSE                                                        45.560177   
PARKROSE                                                        45.560177   
PARKROSE                                                        45.560177   
PARKROSE                                                        45.560177   
SUMNER ASSOCIATION OF NEIGHBORS                                 45.562005   
SUMNER ASSOCIATION OF NEIGHBORS                                 45.562005   
SUMNER ASSOCIATION OF NEIGHBORS                                 45.562005   
ARGAY TERRACE                                                   45.555944   
ARGAY TERRACE                                                   45.555944   
ARGAY TERRACE                                                   45.555944   
ARGAY TERRACE                                                   45.555944   
ARGAY TERRACE                                                   45.555944   
ARGAY TERRACE                                                   45.555944   
ARGAY TERRACE                                                   45.555944   
ARGAY TERRACE                                                   45.555944   
ARGAY TERRACE                                                   45.555944   
ARGAY TERRACE                                                   45.555944   
ARGAY TERRACE                                                   45.555944   
ARGAY TERRACE                                                   45.555944   
WILKES COMMUNITY GROUP                                          45.548804   
WILKES COMMUNITY GROUP                                          45.548804   
WILKES COMMUNITY GROUP                                          45.548804   
WILKES COMMUNITY GROUP                                          45.548804   
SABIN COMMUNITY ASSOCIATION                                     45.552791   
SABIN COMMUNITY ASSOCIATION                                     45.552791   
SABIN COMMUNITY ASSOCIATION                                     45.552791   
SABIN COMMUNITY ASSOCIATION                                     45.552791   
SABIN COMMUNITY ASSOCIATION                     

In [80]:
portland_venues_low.shape

(305, 6)

In [81]:
# display venue counts grouped by neighborhood to see how many neighborhoods I have 
venue_totals_low = portland_venues_low.groupby('Neighborhood').count()
venue_totals_low

Neighborhood Latitude  \
Neighborhood                                                                
ALAMEDA/BEAUMONT-WILSHIRE                                               4   
ARDENWALD-JOHNSON CREEK/WOODSTOCK                                       6   
ARGAY TERRACE                                                          12   
ARGAY/WILKES COMMUNITY GROUP                                            5   
ARNOLD CREEK                                                            2   
ASHCREEK                                                                6   
ASHCREEK/CRESTWOOD                                                      6   
BRENTWOOD-DARLINGTON                                                    4   
BRIDGETON                                                               4   
BRIDLEMILE                                                              3   
BRIDLEMILE/SOUTHWEST HILLS RESIDENTIAL LEAGUE                           1   
BROOKLYN ACTION CORPS                                                   8   
CENTENNIAL COMMUNITY ASSN./PLEASANT VALLEY                              7   
CENTENNIAL COMMUNITY ASSOCIATION                                       10   
COLLINS VIEW                                                            2   
CRESTWOOD                                                               2   
CULLY ASSOCIATION OF NEIGHBORS                                          5   
EAST COLUMBIA                                                           2   
EASTMORELAND                                                            1   
EASTMORELAND/ARDENWALD-JOHNSON CREEK                                    6   
EASTMORELAND/REED                                                       5   
FAR SOUTHWEST                                                           4   
FOREST PARK/LINNTON                                                     1   
FOREST PARK/NORTHWEST DISTRICT ASSOCIATION                              4   
FOSTER-POWELL                                                           5   
GLENFAIR                                                                1   
GRANT PARK                                                              4   
HAYHURST                                                                3   
HEALY HEIGHTS/SOUTHWEST HILLS RESIDENTIAL LEAGUE                        2   
HILLSIDE                                                                9   
HILLSIDE/NORTHWEST DISTRICT ASSN.                                       5   
IRVINGTON COMMUNITY ASSOCIATION                                         5   
KENTON                                                                  3   
LAURELHURST                                                             8   
LENTS                                                                  14   
LENTS/POWELLHURST-GILBERT                                               8   
LINNTON                                                                 3   
MADISON SOUTH                                                           6   
MARKHAM                                                                 7   
MARSHALL PARK                                                           3   
MC UNCLAIMED #11                                                        3   
MC UNCLAIMED #13                                                        3   
MC UNCLAIMED #14                                                        1   
MC UNCLAIMED #5                                                         3   
MILL PARK                                                               4   
MT. TABOR                                                               6   
NORTHWEST HEIGHTS                                                       3   
PARKROSE                                                               13   
PARKROSE HEIGHTS ASSOCIATION OF NEIGHBORS                               4   
PLEASANT VALLEY                                                         2   
PLEASANT VALLEY/POWELLHURST-GILBERT             

In [82]:
venue_totals_low.shape

(65, 6)

In [83]:
# find bakeries. There's only 1
bakeries = portland_venues_low.loc[portland_venues_low['Venue Category'] == 'Bakery']
bakeries

Neighborhood Latitude  \
Neighborhood                                               
ARDENWALD-JOHNSON CREEK/WOODSTOCK              45.462998   

                                   Neighborhood Longitude  \
Neighborhood                                                
ARDENWALD-JOHNSON CREEK/WOODSTOCK              -122.61921   

                                                 Venue  Venue Latitude  \
Neighborhood                                                             
ARDENWALD-JOHNSON CREEK/WOODSTOCK  Franz Bakery Outlet       45.462788   

                                   Venue Longitude Venue Category  
Neighborhood                                                       
ARDENWALD-JOHNSON CREEK/WOODSTOCK      -122.615516         Bakery

In [87]:
# remove the bakery
portland_venues_low = portland_venues_low.drop(['ARDENWALD-JOHNSON CREEK/WOODSTOCK'], axis=0)

KeyError: "['ARDENWALD-JOHNSON CREEK/WOODSTOCK'] not found in axis"

In [88]:
# Display the remaining venue counts, grouped by neighborhood
venue_totals_low = portland_venues_low.groupby('Neighborhood').count()
venue_totals_low

Neighborhood Latitude  \
Neighborhood                                                                
ALAMEDA/BEAUMONT-WILSHIRE                                               4   
ARGAY TERRACE                                                          12   
ARGAY/WILKES COMMUNITY GROUP                                            5   
ARNOLD CREEK                                                            2   
ASHCREEK                                                                6   
ASHCREEK/CRESTWOOD                                                      6   
BRENTWOOD-DARLINGTON                                                    4   
BRIDGETON                                                               4   
BRIDLEMILE                                                              3   
BRIDLEMILE/SOUTHWEST HILLS RESIDENTIAL LEAGUE                           1   
BROOKLYN ACTION CORPS                                                   8   
CENTENNIAL COMMUNITY ASSN./PLEASANT VALLEY                              7   
CENTENNIAL COMMUNITY ASSOCIATION                                       10   
COLLINS VIEW                                                            2   
CRESTWOOD                                                               2   
CULLY ASSOCIATION OF NEIGHBORS                                          5   
EAST COLUMBIA                                                           2   
EASTMORELAND                                                            1   
EASTMORELAND/ARDENWALD-JOHNSON CREEK                                    6   
EASTMORELAND/REED                                                       5   
FAR SOUTHWEST                                                           4   
FOREST PARK/LINNTON                                                     1   
FOREST PARK/NORTHWEST DISTRICT ASSOCIATION                              4   
FOSTER-POWELL                                                           5   
GLENFAIR                                                                1   
GRANT PARK                                                              4   
HAYHURST                                                                3   
HEALY HEIGHTS/SOUTHWEST HILLS RESIDENTIAL LEAGUE                        2   
HILLSIDE                                                                9   
HILLSIDE/NORTHWEST DISTRICT ASSN.                                       5   
IRVINGTON COMMUNITY ASSOCIATION                                         5   
KENTON                                                                  3   
LAURELHURST                                                             8   
LENTS                                                                  14   
LENTS/POWELLHURST-GILBERT                                               8   
LINNTON                                                                 3   
MADISON SOUTH                                                           6   
MARKHAM                                                                 7   
MARSHALL PARK                                                           3   
MC UNCLAIMED #11                                                        3   
MC UNCLAIMED #13                                                        3   
MC UNCLAIMED #14                                                        1   
MC UNCLAIMED #5                                                         3   
MILL PARK                                                               4   
MT. TABOR                                                               6   
NORTHWEST HEIGHTS                                                       3   
PARKROSE                                                               13   
PARKROSE HEIGHTS ASSOCIATION OF NEIGHBORS                               4   
PLEASANT VALLEY                                                         2   
PLEASANT VALLEY/POWELLHURST-GILBERT                                     4   
PORTSMOUTH                                      

In [89]:
venue_totals_low.shape

(64, 6)

In [90]:
portland_venues_low

Neighborhood Latitude  \
Neighborhood                                                                
LINNTON                                                         45.603790   
LINNTON                                                         45.603790   
LINNTON                                                         45.603790   
FOREST PARK/LINNTON                                             45.580630   
UNIVERSITY PARK                                                 45.576354   
UNIVERSITY PARK                                                 45.576354   
UNIVERSITY PARK                                                 45.576354   
UNIVERSITY PARK                                                 45.576354   
MC UNCLAIMED #14                                                45.557689   
CULLY ASSOCIATION OF NEIGHBORS                                  45.563753   
CULLY ASSOCIATION OF NEIGHBORS                                  45.563753   
CULLY ASSOCIATION OF NEIGHBORS                                  45.563753   
CULLY ASSOCIATION OF NEIGHBORS                                  45.563753   
CULLY ASSOCIATION OF NEIGHBORS                                  45.563753   
PARKROSE                                                        45.560177   
PARKROSE                                                        45.560177   
PARKROSE                                                        45.560177   
PARKROSE                                                        45.560177   
PARKROSE                                                        45.560177   
PARKROSE                                                        45.560177   
PARKROSE                                                        45.560177   
PARKROSE                                                        45.560177   
PARKROSE                                                        45.560177   
PARKROSE                                                        45.560177   
PARKROSE                                                        45.560177   
PARKROSE                                                        45.560177   
PARKROSE                                                        45.560177   
SUMNER ASSOCIATION OF NEIGHBORS                                 45.562005   
SUMNER ASSOCIATION OF NEIGHBORS                                 45.562005   
SUMNER ASSOCIATION OF NEIGHBORS                                 45.562005   
ARGAY TERRACE                                                   45.555944   
ARGAY TERRACE                                                   45.555944   
ARGAY TERRACE                                                   45.555944   
ARGAY TERRACE                                                   45.555944   
ARGAY TERRACE                                                   45.555944   
ARGAY TERRACE                                                   45.555944   
ARGAY TERRACE                                                   45.555944   
ARGAY TERRACE                                                   45.555944   
ARGAY TERRACE                                                   45.555944   
ARGAY TERRACE                                                   45.555944   
ARGAY TERRACE                                                   45.555944   
ARGAY TERRACE                                                   45.555944   
WILKES COMMUNITY GROUP                                          45.548804   
WILKES COMMUNITY GROUP                                          45.548804   
WILKES COMMUNITY GROUP                                          45.548804   
WILKES COMMUNITY GROUP                                          45.548804   
SABIN COMMUNITY ASSOCIATION                                     45.552791   
SABIN COMMUNITY ASSOCIATION                                     45.552791   
SABIN COMMUNITY ASSOCIATION                                     45.552791   
SABIN COMMUNITY ASSOCIATION                                     45.552791   
SABIN COMMUNITY ASSOCIATION                     

#### I have removed busy nieghborhoods, and neighborhoods that already have coffee shops or bakeries. Now I would like to find neighborhoods that have at least one park. 

In [91]:
# find neighborhoods with parks.
parks = portland_venues_low.loc[portland_venues_low['Venue Category'] == 'Park']
parks

Neighborhood Latitude  \
Neighborhood                                                              
NORTHWEST HEIGHTS                                             45.540300   
MADISON SOUTH                                                 45.541545   
ARGAY/WILKES COMMUNITY GROUP                                  45.550552   
PARKROSE HEIGHTS ASSOCIATION OF NEIGHBORS                     45.540394   
GRANT PARK                                                    45.539315   
HILLSIDE/NORTHWEST DISTRICT ASSN.                             45.528865   
HILLSIDE                                                      45.525879   
HILLSIDE                                                      45.525879   
MT. TABOR                                                     45.514310   
MILL PARK                                                     45.512087   
BRIDLEMILE                                                    45.491312   
FOSTER-POWELL                                                 45.492390   
LENTS/POWELLHURST-GILBERT                                     45.491147   
HEALY HEIGHTS/SOUTHWEST HILLS RESIDENTIAL LEAGUE              45.492353   
CENTENNIAL COMMUNITY ASSN./PLEASANT VALLEY                    45.490246   
PLEASANT VALLEY                                               45.472157   
SOUTHWEST HILLS RESIDENTIAL LEAGUE                            45.490485   
HAYHURST                                                      45.481040   
EASTMORELAND                                                  45.473030   
BRENTWOOD-DARLINGTON                                          45.468852   
MC UNCLAIMED #11                                              45.459578   
MC UNCLAIMED #11                                              45.459578   
WEST PORTLAND PARK                                            45.446965   
WEST PORTLAND PARK                                            45.446965   
KENTON                                                        45.594651   
EAST COLUMBIA                                                 45.593898   
PORTSMOUTH                                                    45.587194   

                                                  Neighborhood Longitude  \
Neighborhood                                                               
NORTHWEST HEIGHTS                                            -122.771887   
MADISON SOUTH                                                -122.574206   
ARGAY/WILKES COMMUNITY GROUP                                 -122.510636   
PARKROSE HEIGHTS ASSOCIATION OF NEIGHBORS                    -122.548186   
GRANT PARK                                                   -122.629179   
HILLSIDE/NORTHWEST DISTRICT ASSN.                            -122.705431   
HILLSIDE                                                     -122.715946   
HILLSIDE                                                     -122.715946   
MT. TABOR                                                    -122.598677   
MILL PARK                                                    -122.540271   
BRIDLEMILE                                                   -122.726829   
FOSTER-POWELL                                                -122.590429   
LENTS/POWELLHURST-GILBERT                                    -122.554980   
HEALY HEIGHTS/SOUTHWEST HILLS RESIDENTIAL LEAGUE             -122.698403   
CENTENNIAL COMMUNITY ASSN./PLEASANT VALLEY                   -122.492408   
PLEASANT VALLEY                                              -122.506524   
SOUTHWEST HILLS RESIDENTIAL LEAGUE                           -122.694513   
HAYHURST                                                     -122.727770   
EASTMORELAND                                                 -122.631509   
BRENTWOOD-DARLINGTON                                         -122.597273   
MC UNCLAIMED #11                                             -122.669907   
MC UNCLAIMED #11                                             -122.669907   
WEST PORTLAND PARK                                

In [92]:
# get list of neighborhoods with parks
parks_list = parks.index.values
parks_list

array(['NORTHWEST HEIGHTS', 'MADISON SOUTH',
       'ARGAY/WILKES COMMUNITY GROUP',
       'PARKROSE HEIGHTS ASSOCIATION OF NEIGHBORS', 'GRANT PARK',
       'HILLSIDE/NORTHWEST DISTRICT ASSN.', 'HILLSIDE', 'HILLSIDE',
       'MT. TABOR', 'MILL PARK', 'BRIDLEMILE', 'FOSTER-POWELL',
       'LENTS/POWELLHURST-GILBERT',
       'HEALY HEIGHTS/SOUTHWEST HILLS RESIDENTIAL LEAGUE',
       'CENTENNIAL COMMUNITY ASSN./PLEASANT VALLEY', 'PLEASANT VALLEY',
       'SOUTHWEST HILLS RESIDENTIAL LEAGUE', 'HAYHURST', 'EASTMORELAND',
       'BRENTWOOD-DARLINGTON', 'MC UNCLAIMED #11', 'MC UNCLAIMED #11',
       'WEST PORTLAND PARK', 'WEST PORTLAND PARK', 'KENTON',
       'EAST COLUMBIA', 'PORTSMOUTH'], dtype=object)

In [95]:
Portland_venues.reset_index(level =['Neighborhood'], inplace = True) 
Portland_venues.head()

KeyError: 'Requested level (Neighborhood) does not match index name (None)'

In [96]:
Portland_venues.head()

Neighborhood  Neighborhood Latitude  Neighborhood Longitude  \
0              LINNTON              45.603790             -122.793264   
1              LINNTON              45.603790             -122.793264   
2              LINNTON              45.603790             -122.793264   
3  FOREST PARK/LINNTON              45.580630             -122.781775   
4       CATHEDRAL PARK              45.587368             -122.757317   

                            Venue  Venue Latitude  Venue Longitude  \
0                          Subway       45.603014      -122.788590   
1                           Shell       45.603473      -122.788861   
2                        7-Eleven       45.602668      -122.788165   
3  Forest Park Hardesty Trailhead       45.578979      -122.781064   
4                  Cathedral Park       45.587744      -122.759822   

      Venue Category  
0     Sandwich Place  
1        Gas Station  
2  Convenience Store  
3              Trail  
4               Park

#### Get subset of Portland_venues that only has neighborhoods that aren't busy (less than 15 venues), have no coffee shops or bakeries, and have at least one park

In [97]:
parks = Portland_venues[Portland_venues['Neighborhood'].isin(['NORTHWEST HEIGHTS', 'MADISON SOUTH',
       'ARGAY/WILKES COMMUNITY GROUP',
       'PARKROSE HEIGHTS ASSOCIATION OF NEIGHBORS', 'GRANT PARK',
       'HILLSIDE/NORTHWEST DISTRICT ASSN.', 'HILLSIDE', 'HILLSIDE',
       'MT. TABOR', 'MILL PARK', 'BRIDLEMILE', 'FOSTER-POWELL',
       'LENTS/POWELLHURST-GILBERT',
       'HEALY HEIGHTS/SOUTHWEST HILLS RESIDENTIAL LEAGUE',
       'CENTENNIAL COMMUNITY ASSN./PLEASANT VALLEY', 'PLEASANT VALLEY',
       'SOUTHWEST HILLS RESIDENTIAL LEAGUE', 'HAYHURST', 'EASTMORELAND',
       'BRENTWOOD-DARLINGTON', 'MC UNCLAIMED #11', 'MC UNCLAIMED #11',
       'WEST PORTLAND PARK', 'WEST PORTLAND PARK', 'KENTON',
       'EAST COLUMBIA', 'PORTSMOUTH'])]


In [98]:
parks

Neighborhood  Neighborhood Latitude  \
417                                  NORTHWEST HEIGHTS              45.540300   
418                                  NORTHWEST HEIGHTS              45.540300   
419                                  NORTHWEST HEIGHTS              45.540300   
449                                      MADISON SOUTH              45.541545   
450                                      MADISON SOUTH              45.541545   
451                                      MADISON SOUTH              45.541545   
452                                      MADISON SOUTH              45.541545   
453                                      MADISON SOUTH              45.541545   
454                                      MADISON SOUTH              45.541545   
455                       ARGAY/WILKES COMMUNITY GROUP              45.550552   
456                       ARGAY/WILKES COMMUNITY GROUP              45.550552   
457                       ARGAY/WILKES COMMUNITY GROUP              45.550552   
458                       ARGAY/WILKES COMMUNITY GROUP              45.550552   
459                       ARGAY/WILKES COMMUNITY GROUP              45.550552   
555          PARKROSE HEIGHTS ASSOCIATION OF NEIGHBORS              45.540394   
556          PARKROSE HEIGHTS ASSOCIATION OF NEIGHBORS              45.540394   
557          PARKROSE HEIGHTS ASSOCIATION OF NEIGHBORS              45.540394   
558          PARKROSE HEIGHTS ASSOCIATION OF NEIGHBORS              45.540394   
639                                         GRANT PARK              45.539315   
640                                         GRANT PARK              45.539315   
641                                         GRANT PARK              45.539315   
642                                         GRANT PARK              45.539315   
1008                 HILLSIDE/NORTHWEST DISTRICT ASSN.              45.528865   
1009                 HILLSIDE/NORTHWEST DISTRICT ASSN.              45.528865   
1010                 HILLSIDE/NORTHWEST DISTRICT ASSN.              45.528865   
1011                 HILLSIDE/NORTHWEST DISTRICT ASSN.              45.528865   
1012                 HILLSIDE/NORTHWEST DISTRICT ASSN.              45.528865   
1076                                          HILLSIDE              45.525879   
1077                                          HILLSIDE              45.525879   
1078                                          HILLSIDE              45.525879   
1079                                          HILLSIDE              45.525879   
1080                                          HILLSIDE              45.525879   
1081                                          HILLSIDE              45.525879   
1082                                          HILLSIDE              45.525879   
1083                                          HILLSIDE              45.525879   
1084                                          HILLSIDE              45.525879   
1402                                         MT. TABOR              45.514310   
1403                                         MT. TABOR              45.514310   
1404                                         MT. TABOR              45.514310   
1405                                         MT. TABOR              45.514310   
1406                                         MT. TABOR              45.514310   
1407                                         MT. TABOR              45.514310   
1425                SOUTHWEST HILLS RESIDENTIAL LEAGUE              45.503673   
1426                SOUTHWEST HILLS RESIDENTIAL LEAGUE              45.503673   
1427                SOUTHWEST HILLS RESIDENTIAL LEAGUE              45.503673   
1428                SOUTHWEST HILLS RESIDENTIAL LEAGUE              45.503673   
1429                SOUTHWEST HILLS RESIDENTIAL LEAGUE              45.503673   
1430                SOUTHWEST HILLS RESIDENTIAL LEAGUE              45.503673   
1559                                         MILL PARK              45.51

In [99]:
# Display the remaining venue counts, grouped by neighborhood
parks_total = parks.groupby('Neighborhood').count()
parks_total

Neighborhood Latitude  \
Neighborhood                                                              
ARGAY/WILKES COMMUNITY GROUP                                          5   
BRENTWOOD-DARLINGTON                                                  4   
BRIDLEMILE                                                            3   
CENTENNIAL COMMUNITY ASSN./PLEASANT VALLEY                            7   
EAST COLUMBIA                                                         2   
EASTMORELAND                                                          1   
FOSTER-POWELL                                                         5   
GRANT PARK                                                            4   
HAYHURST                                                              3   
HEALY HEIGHTS/SOUTHWEST HILLS RESIDENTIAL LEAGUE                      2   
HILLSIDE                                                              9   
HILLSIDE/NORTHWEST DISTRICT ASSN.                                     5   
KENTON                                                                3   
LENTS/POWELLHURST-GILBERT                                             8   
MADISON SOUTH                                                         6   
MC UNCLAIMED #11                                                      3   
MILL PARK                                                             4   
MT. TABOR                                                             6   
NORTHWEST HEIGHTS                                                     3   
PARKROSE HEIGHTS ASSOCIATION OF NEIGHBORS                             4   
PLEASANT VALLEY                                                       2   
PORTSMOUTH                                                            4   
SOUTHWEST HILLS RESIDENTIAL LEAGUE                                    9   
WEST PORTLAND PARK                                                    4   

                                                  Neighborhood Longitude  \
Neighborhood                                                               
ARGAY/WILKES COMMUNITY GROUP                                           5   
BRENTWOOD-DARLINGTON                                                   4   
BRIDLEMILE                                                             3   
CENTENNIAL COMMUNITY ASSN./PLEASANT VALLEY                             7   
EAST COLUMBIA                                                          2   
EASTMORELAND                                                           1   
FOSTER-POWELL                                                          5   
GRANT PARK                                                             4   
HAYHURST                                                               3   
HEALY HEIGHTS/SOUTHWEST HILLS RESIDENTIAL LEAGUE                       2   
HILLSIDE                                                               9   
HILLSIDE/NORTHWEST DISTRICT ASSN.                                      5   
KENTON                                                                 3   
LENTS/POWELLHURST-GILBERT                                              8   
MADISON SOUTH                                                          6   
MC UNCLAIMED #11                                                       3   
MILL PARK                                                              4   
MT. TABOR                                                              6   
NORTHWEST HEIGHTS                                                      3   
PARKROSE HEIGHTS ASSOCIATION OF NEIGHBORS                              4   
PLEASANT VALLEY                                                        2   
PORTSMOUTH                                                             4   
SOUTHWEST HILLS RESIDENTIAL LEAGUE                                     9   
WEST PORTLAND PARK                                                     4   

                                                  Venue  Venue Latitude  \
Neighborhood                                   

In [100]:
parks_total.shape

(24, 6)

In [102]:
parks

Neighborhood  Neighborhood Latitude  \
417                                  NORTHWEST HEIGHTS              45.540300   
418                                  NORTHWEST HEIGHTS              45.540300   
419                                  NORTHWEST HEIGHTS              45.540300   
449                                      MADISON SOUTH              45.541545   
450                                      MADISON SOUTH              45.541545   
451                                      MADISON SOUTH              45.541545   
452                                      MADISON SOUTH              45.541545   
453                                      MADISON SOUTH              45.541545   
454                                      MADISON SOUTH              45.541545   
455                       ARGAY/WILKES COMMUNITY GROUP              45.550552   
456                       ARGAY/WILKES COMMUNITY GROUP              45.550552   
457                       ARGAY/WILKES COMMUNITY GROUP              45.550552   
458                       ARGAY/WILKES COMMUNITY GROUP              45.550552   
459                       ARGAY/WILKES COMMUNITY GROUP              45.550552   
555          PARKROSE HEIGHTS ASSOCIATION OF NEIGHBORS              45.540394   
556          PARKROSE HEIGHTS ASSOCIATION OF NEIGHBORS              45.540394   
557          PARKROSE HEIGHTS ASSOCIATION OF NEIGHBORS              45.540394   
558          PARKROSE HEIGHTS ASSOCIATION OF NEIGHBORS              45.540394   
639                                         GRANT PARK              45.539315   
640                                         GRANT PARK              45.539315   
641                                         GRANT PARK              45.539315   
642                                         GRANT PARK              45.539315   
1008                 HILLSIDE/NORTHWEST DISTRICT ASSN.              45.528865   
1009                 HILLSIDE/NORTHWEST DISTRICT ASSN.              45.528865   
1010                 HILLSIDE/NORTHWEST DISTRICT ASSN.              45.528865   
1011                 HILLSIDE/NORTHWEST DISTRICT ASSN.              45.528865   
1012                 HILLSIDE/NORTHWEST DISTRICT ASSN.              45.528865   
1076                                          HILLSIDE              45.525879   
1077                                          HILLSIDE              45.525879   
1078                                          HILLSIDE              45.525879   
1079                                          HILLSIDE              45.525879   
1080                                          HILLSIDE              45.525879   
1081                                          HILLSIDE              45.525879   
1082                                          HILLSIDE              45.525879   
1083                                          HILLSIDE              45.525879   
1084                                          HILLSIDE              45.525879   
1402                                         MT. TABOR              45.514310   
1403                                         MT. TABOR              45.514310   
1404                                         MT. TABOR              45.514310   
1405                                         MT. TABOR              45.514310   
1406                                         MT. TABOR              45.514310   
1407                                         MT. TABOR              45.514310   
1425                SOUTHWEST HILLS RESIDENTIAL LEAGUE              45.503673   
1426                SOUTHWEST HILLS RESIDENTIAL LEAGUE              45.503673   
1427                SOUTHWEST HILLS RESIDENTIAL LEAGUE              45.503673   
1428                SOUTHWEST HILLS RESIDENTIAL LEAGUE              45.503673   
1429                SOUTHWEST HILLS RESIDENTIAL LEAGUE              45.503673   
1430                SOUTHWEST HILLS RESIDENTIAL LEAGUE              45.503673   
1559                                         MILL PARK              45.51

In [103]:
parks = parks.set_index("Neighborhood")

#### Now that I've narrowed my list down to 24 neighborhoods, I looked at them individually. I need to drop a few more rows for some special cases (e.g. a neighborhood with venues that all consist of a zoo, one has a food truck that is a coffee shop, one has a freightliner delaership, one with a museum, and one with a big intersection/construction site.

In [104]:
# drop rows with coffee shops
final_neighborhoods = parks.drop(['HILLSIDE','SOUTHWEST HILLS RESIDENTIAL LEAGUE', 
                                                'KENTON','EAST COLUMBIA', 'MC UNCLAIMED #11', 
                                                'PLEASANT VALLEY', 'HILLSIDE/NORTHWEST DISTRICT ASSN.'], axis=0)

In [105]:
final_neighborhoods

Neighborhood Latitude  \
Neighborhood                                                              
NORTHWEST HEIGHTS                                             45.540300   
NORTHWEST HEIGHTS                                             45.540300   
NORTHWEST HEIGHTS                                             45.540300   
MADISON SOUTH                                                 45.541545   
MADISON SOUTH                                                 45.541545   
MADISON SOUTH                                                 45.541545   
MADISON SOUTH                                                 45.541545   
MADISON SOUTH                                                 45.541545   
MADISON SOUTH                                                 45.541545   
ARGAY/WILKES COMMUNITY GROUP                                  45.550552   
ARGAY/WILKES COMMUNITY GROUP                                  45.550552   
ARGAY/WILKES COMMUNITY GROUP                                  45.550552   
ARGAY/WILKES COMMUNITY GROUP                                  45.550552   
ARGAY/WILKES COMMUNITY GROUP                                  45.550552   
PARKROSE HEIGHTS ASSOCIATION OF NEIGHBORS                     45.540394   
PARKROSE HEIGHTS ASSOCIATION OF NEIGHBORS                     45.540394   
PARKROSE HEIGHTS ASSOCIATION OF NEIGHBORS                     45.540394   
PARKROSE HEIGHTS ASSOCIATION OF NEIGHBORS                     45.540394   
GRANT PARK                                                    45.539315   
GRANT PARK                                                    45.539315   
GRANT PARK                                                    45.539315   
GRANT PARK                                                    45.539315   
MT. TABOR                                                     45.514310   
MT. TABOR                                                     45.514310   
MT. TABOR                                                     45.514310   
MT. TABOR                                                     45.514310   
MT. TABOR                                                     45.514310   
MT. TABOR                                                     45.514310   
MILL PARK                                                     45.512087   
MILL PARK                                                     45.512087   
MILL PARK                                                     45.512087   
MILL PARK                                                     45.512087   
BRIDLEMILE                                                    45.491312   
BRIDLEMILE                                                    45.491312   
BRIDLEMILE                                                    45.491312   
FOSTER-POWELL                                                 45.492390   
FOSTER-POWELL                                                 45.492390   
FOSTER-POWELL                                                 45.492390   
FOSTER-POWELL                                                 45.492390   
FOSTER-POWELL                                                 45.492390   
LENTS/POWELLHURST-GILBERT                                     45.491147   
LENTS/POWELLHURST-GILBERT                                     45.491147   
LENTS/POWELLHURST-GILBERT                                     45.491147   
LENTS/POWELLHURST-GILBERT                                     45.491147   
LENTS/POWELLHURST-GILBERT                                     45.491147   
LENTS/POWELLHURST-GILBERT                                     45.491147   
LENTS/POWELLHURST-GILBERT                                     45.491147   
LENTS/POWELLHURST-GILBERT                                     45.491147   
HEALY HEIGHTS/SOUTHWEST HILLS RESIDENTIAL LEAGUE              45.492353   
HEALY HEIGHTS/SOUTHWEST HILLS RESIDENTIAL LEAGUE              45.492353   
CENTENNIAL COMMUNITY ASSN./PLEASANT VALLEY                    45.490246   
CENTENNIAL COMMUNITY ASSN./PLEASANT VALLEY                    45.490246   


#### I've narrowed it down to 17 neighborhoods (down from 130)

In [106]:
# Display the remaining venue counts, grouped by neighborhood
final_neighborhoods_grouped = final_neighborhoods.groupby('Neighborhood').count()
final_neighborhoods_grouped 

Neighborhood Latitude  \
Neighborhood                                                              
ARGAY/WILKES COMMUNITY GROUP                                          5   
BRENTWOOD-DARLINGTON                                                  4   
BRIDLEMILE                                                            3   
CENTENNIAL COMMUNITY ASSN./PLEASANT VALLEY                            7   
EASTMORELAND                                                          1   
FOSTER-POWELL                                                         5   
GRANT PARK                                                            4   
HAYHURST                                                              3   
HEALY HEIGHTS/SOUTHWEST HILLS RESIDENTIAL LEAGUE                      2   
LENTS/POWELLHURST-GILBERT                                             8   
MADISON SOUTH                                                         6   
MILL PARK                                                             4   
MT. TABOR                                                             6   
NORTHWEST HEIGHTS                                                     3   
PARKROSE HEIGHTS ASSOCIATION OF NEIGHBORS                             4   
PORTSMOUTH                                                            4   
WEST PORTLAND PARK                                                    4   

                                                  Neighborhood Longitude  \
Neighborhood                                                               
ARGAY/WILKES COMMUNITY GROUP                                           5   
BRENTWOOD-DARLINGTON                                                   4   
BRIDLEMILE                                                             3   
CENTENNIAL COMMUNITY ASSN./PLEASANT VALLEY                             7   
EASTMORELAND                                                           1   
FOSTER-POWELL                                                          5   
GRANT PARK                                                             4   
HAYHURST                                                               3   
HEALY HEIGHTS/SOUTHWEST HILLS RESIDENTIAL LEAGUE                       2   
LENTS/POWELLHURST-GILBERT                                              8   
MADISON SOUTH                                                          6   
MILL PARK                                                              4   
MT. TABOR                                                              6   
NORTHWEST HEIGHTS                                                      3   
PARKROSE HEIGHTS ASSOCIATION OF NEIGHBORS                              4   
PORTSMOUTH                                                             4   
WEST PORTLAND PARK                                                     4   

                                                  Venue  Venue Latitude  \
Neighborhood                                                              
ARGAY/WILKES COMMUNITY GROUP                          5               5   
BRENTWOOD-DARLINGTON                                  4               4   
BRIDLEMILE                                            3               3   
CENTENNIAL COMMUNITY ASSN./PLEASANT VALLEY            7               7   
EASTMORELAND                                          1               1   
FOSTER-POWELL                                         5               5   
GRANT PARK                                            4               4   
HAYHURST                                              3               3   
HEALY HEIGHTS/SOUTHWEST HILLS RESIDENTIAL LEAGUE      2               2   
LENTS/POWELLHURST-GILBERT                             8               8   
MADISON SOUTH                                         6               6   
MILL PARK                                             4               4   
MT. TABOR                                             6               6   
NORTHWEST HEIGHTS                                     

### Create a final map showing ideal locations for a bakery/coffee shop

In [108]:
# get list of neighborhoods with parks
final_neighborhoods_list = final_neighborhoods_grouped.index.values
final_neighborhoods_list

array(['ARGAY/WILKES COMMUNITY GROUP', 'BRENTWOOD-DARLINGTON',
       'BRIDLEMILE', 'CENTENNIAL COMMUNITY ASSN./PLEASANT VALLEY',
       'EASTMORELAND', 'FOSTER-POWELL', 'GRANT PARK', 'HAYHURST',
       'HEALY HEIGHTS/SOUTHWEST HILLS RESIDENTIAL LEAGUE',
       'LENTS/POWELLHURST-GILBERT', 'MADISON SOUTH', 'MILL PARK',
       'MT. TABOR', 'NORTHWEST HEIGHTS',
       'PARKROSE HEIGHTS ASSOCIATION OF NEIGHBORS', 'PORTSMOUTH',
       'WEST PORTLAND PARK'], dtype=object)

In [110]:
df.head()

NAME   Latitude   Longitude
0              LINNTON  45.603790 -122.793264
1  FOREST PARK/LINNTON  45.580630 -122.781775
2          FOREST PARK  45.564383 -122.792078
3       CATHEDRAL PARK  45.587368 -122.757317
4      UNIVERSITY PARK  45.576354 -122.730079

#### Need to get a df with lat and long, so I went back to the original list (df) and got a subset with my chosen neighborhoods

In [113]:
df_final = df[df['NAME'].isin(['ARGAY/WILKES COMMUNITY GROUP', 'BRENTWOOD-DARLINGTON',
       'BRIDLEMILE', 'CENTENNIAL COMMUNITY ASSN./PLEASANT VALLEY',
       'EASTMORELAND', 'FOSTER-POWELL', 'GRANT PARK', 'HAYHURST',
       'HEALY HEIGHTS/SOUTHWEST HILLS RESIDENTIAL LEAGUE',
       'LENTS/POWELLHURST-GILBERT', 'MADISON SOUTH', 'MILL PARK',
       'MT. TABOR', 'NORTHWEST HEIGHTS',
       'PARKROSE HEIGHTS ASSOCIATION OF NEIGHBORS', 'PORTSMOUTH',
       'WEST PORTLAND PARK'])]

In [114]:
df_final

NAME   Latitude   Longitude
23                                  NORTHWEST HEIGHTS  45.540300 -122.771887
25                                      MADISON SOUTH  45.541545 -122.574206
26                       ARGAY/WILKES COMMUNITY GROUP  45.550552 -122.510636
33          PARKROSE HEIGHTS ASSOCIATION OF NEIGHBORS  45.540394 -122.548186
39                                         GRANT PARK  45.539315 -122.629179
63                                          MT. TABOR  45.514310 -122.598677
71                                          MILL PARK  45.512087 -122.540271
81                                         BRIDLEMILE  45.491312 -122.726829
84                                      FOSTER-POWELL  45.492390 -122.590429
86                          LENTS/POWELLHURST-GILBERT  45.491147 -122.554980
87   HEALY HEIGHTS/SOUTHWEST HILLS RESIDENTIAL LEAGUE  45.492353 -122.698403
88         CENTENNIAL COMMUNITY ASSN./PLEASANT VALLEY  45.490246 -122.492408
95                                           HAYHURST  45.481040 -122.727770
97                                       EASTMORELAND  45.473030 -122.631509
101                              BRENTWOOD-DARLINGTON  45.468852 -122.597273
113                                WEST PORTLAND PARK  45.446965 -122.720507
129                                        PORTSMOUTH  45.587194 -122.718379

#### reset the index

In [116]:
df_final.reset_index(drop=True, inplace=True)
df_final

NAME   Latitude   Longitude
0                                  NORTHWEST HEIGHTS  45.540300 -122.771887
1                                      MADISON SOUTH  45.541545 -122.574206
2                       ARGAY/WILKES COMMUNITY GROUP  45.550552 -122.510636
3          PARKROSE HEIGHTS ASSOCIATION OF NEIGHBORS  45.540394 -122.548186
4                                         GRANT PARK  45.539315 -122.629179
5                                          MT. TABOR  45.514310 -122.598677
6                                          MILL PARK  45.512087 -122.540271
7                                         BRIDLEMILE  45.491312 -122.726829
8                                      FOSTER-POWELL  45.492390 -122.590429
9                          LENTS/POWELLHURST-GILBERT  45.491147 -122.554980
10  HEALY HEIGHTS/SOUTHWEST HILLS RESIDENTIAL LEAGUE  45.492353 -122.698403
11        CENTENNIAL COMMUNITY ASSN./PLEASANT VALLEY  45.490246 -122.492408
12                                          HAYHURST  45.481040 -122.727770
13                                      EASTMORELAND  45.473030 -122.631509
14                              BRENTWOOD-DARLINGTON  45.468852 -122.597273
15                                WEST PORTLAND PARK  45.446965 -122.720507
16                                        PORTSMOUTH  45.587194 -122.718379

In [115]:
# create final map of narrowed down Portland neighborhoods using latitude and longitude values
m_final = folium.Map(location=[45.5236, -122.6750], zoom_start=12)

# add markers to map
for lat, lng, label in zip(df_final['Latitude'], df_final['Longitude'], df_final['NAME']):
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(m_final)  
    
m_final